netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -8,7 +8,61 @@
 
 #define INITGUID
 
-#include "net_ebpf_ext.h"
+#include "net_ebpf_ext_xdp.h"
+
+//
+// Utility functions.
+//
+NTSTATUS
+_net_ebpf_extension_xdp_validate_if_index(uint32_t if_index)
+{
+    NTSTATUS status = STATUS_SUCCESS;
+    PMIB_IF_TABLE2 table;
+    ULONG i;
+    bool valid_if_index = FALSE;
+
+    status = GetIfTable2(&table);
+    if (!NT_SUCCESS(status))
+        goto Exit;
+
+    for (i = 0; i < table->NumEntries; i++) {
+        if (table->Table[i].InterfaceIndex == if_index) {
+            valid_if_index = TRUE;
+            break;
+        }
+    }
+
+    if (!valid_if_index)
+        status = STATUS_INVALID_PARAMETER;
```

Comment:
I think you can do this more efficiently by calling GetIfEntry2 (https://docs.microsoft.com/en-us/windows/win32/api/netioapi/nf-netioapi-getifentry2) instead of GetIfTable2.   If you use GetIfEntry2 then I don't think you need a loop or a free.

Reply:
Done.